### ISSUE TITLE
Database encryption does not work when signed in

### ISSUE THREAD

**Issue by Funman2** [nunchuk-io/nunchuk-desktop#20]
In windows app, when signed in to a nunchuk account, I go to settings and select encryption, i can set a passphrase for signed in account, the encryption gets set, keys get encrypted, but when I sign out/in in again it does not promt the passphrase for encryption, then it can not read the keys anymore.

When in guest mode, encryption works fine! It asks for passphrase and key data is stored in an encrypted way in the folder App data roaming..

Otherwise the only way to use nunchuk is my seedwords get stored in clear text! That is not an option at all, cause you have a digital copy on windows accessible for all apps!

**Comment by Funman2** [comment]
Also in both modes guest and signed in mode the setting encryption set to -on is getting lost when leaving the settings menu, but despite that the encryption stays activated in guest mode, when signend in , it does not activate encryption again after once set...

**Comment by tongvanlinh** [comment]
@Funman2 thank you, I think I missed this in online mode after login, let me check and feedback to you. (there is tip that you can change login password same with you encrytion passphrase, then I think it could decrypt your db after login)

**Comment by Funman2** [comment]
Yes that makes sense,  because another issue I detected: when I previously log in to guest account and then sign in to nunchuk account I can see the keys from guest, probably because it's the exact same password...so setting it differently should also solve that for now,  I want to strictly separate guest and online mode. 
Will try this and get back to you when back at home! 
Walking the dog right now;-)

**Comment by Funman2** [comment]
Hey, setting the login password and the encryption passphrase the same, did not resolve the issue.

The encryption does not work when signed in. It just encrypts the key, but never decrypt it again. also after first attemt of enabling the encryption i can never change the passphrase for encryption, instead i get this error:
![screenshot nunchuk](https://user-images.githubusercontent.com/105175715/182578967-03c2e119-3194-423f-9bcb-0010e225e2eb.jpg)

But i could resolve my keys from guest mode beeing displayed in signed in mode by using different passwords for login and encryption.

Also when siging in and selecting the stay signed in option, it never stores the login, have to retype my credentials every time...

**Comment by tongvanlinh** [comment]
@Funman2 i see, let me fix it in notify you later

**Comment by tongvanlinh** [comment]
@Funman2 Could you check this on the pre-release version [here](https://github.com/nunchuk-io/nunchuk-desktop/releases/tag/1.9.12).
If it has any issue, please let me know

**Comment by Funman2** [comment]
Thx, looking forward. Will get back to you asap!

**Comment by Funman2** [comment]
Hey, sorry to say, Icould only try on windows yet, but I guess this version you posted should resolve the encryption issues?
Unfortionatly I get all the same errors for the encryption as before, encryption works fine in guest mode.
When signing in, it still always does not remember the stay signed in setting first of all, so i always have to retype the credentials all over, which gets a little tyring by now...then after I set up encryption for signed in mode(different passphrase then for guest) it does not let me sign in any more, instead I get the following screens:
![Signed in 1 encryption](https://user-images.githubusercontent.com/105175715/184506169-63c3a9b0-9635-43e9-b0ac-a2e1b32f8a5c.png)
![Signed in 2](https://user-images.githubusercontent.com/105175715/184506170-40d355f5-08fa-4e6a-9a25-270824efefe4.jpg)
:
Can not sign in at all anymore now...

**Comment by Funman2** [comment]
Update, also setting login password the same as encryption passphrase did not help to solve, also still the screen to enter pasphrase does not pop up in account mode (to 'enter the dragon`)

**Comment by tongvanlinh** [comment]
@Funman2 are you in testnet or signet or main net ?

**Comment by Funman2** [comment]
Mainnet.

**Comment by tongvanlinh** [comment]
@Funman2 let me check more on mainnet.

**Comment by tongvanlinh** [comment]
@Funman2 One more question. Is there any wallet/key in  online mode ? Because if there is no data (wallet/key), it wont promt input passphrase, until db existed

**Comment by Funman2** [comment]
Yes there was key data in the account, it used to have one software key, now it got deleted after first attempt failed, so will reset it up and try again to double check, will get back to u in a minute....

**Comment by Funman2** [comment]
i just resetup the key, changed account password to be sure, then set encryption passward to the same as login passward as you recommended once to give it one more shot, it did not help unfortionatly same result: no prompting for encryption passward.
Also i would like to ad that i noticed, if i set up encryption the selector for encryption always switches back to toggle setting off, after setting encryption. It still works in guest mode although the app indicates switch shows encryption turned off...

**Comment by tongvanlinh** [comment]
@Funman2 thx, I'm debuging based on your info. Thank you, I will notify you soon

**Comment by tongvanlinh** [comment]
@Funman2  could you try [this](https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.12/nunchuk-windows-1.9.12.zip) once again. It's just my guessing point, it need to be improved more

**Comment by Funman2** [comment]
Yes will do! Busy right now,  I'll get back asap.

**Comment by Funman2** [comment]
Hey man, don't know what you did but, YES that one did the trick! Awesome, it worked, prompted for encryption...! 
After entering encryption password I could enter the account and everything showed up!
Thanks for your consitent efforts, finally I feel more save with my seedwords stored on the device! Great work.

**Comment by Funman2** [comment]
For linux I will have to set up my VM again to test,  will do that tonight but as far as I know I could not set up the encryption at all even in guest mode in my linux ubuntu distribution(20.4), not sure if this was linux isue or VM related tbh, will try and get back...
Also if I find the time I will look into setting up dual boot system with linux.

**Comment by Funman2** [comment]
Now on the encryption settings it keeps the toggle encryption on setting in memory as well. So this is great improvement!

Now for next version it would be really nice if it also would accept the stay signend in option for Nunchuk account, so i would not have to reenter credentials every time opening the app. But encryption password should always be asked if set to enccryption of course as a savety measure I think.

**Comment by Funman2** [comment]
I saw u even fixed the missing transcations in the "view all transactions" tab...all transactions from last year show up now! Great job!

small minor improvement would be after entering encryption password that you  would be able to just hit enter after entering passphrase, that does'nt work right now, you would have to click.
Also i realized when entering a btc address it´s possible to copy and past by keyboard shortcuts but some usere may need a right click dropdown menue wich is not working for copy/paste action in nunchuk desktop (all systems)right now. But thats very small issues. Otherwise you are really getting there now! Great work!

**Comment by Funman2** [comment]
One more little thing: when signed in, then signed out, then guest mode it partially keeps my account information in guest mode for instance profile pic and name...

**Comment by tongvanlinh** [comment]
@Funman2 thx you. I'm really impressed with your contribution. I'll improve our product with your suggest and comments. When we have official release I will notify you. It will be soon

**Comment by Funman2** [comment]
I have to thank you for creating nunchuk and constantly improving it! Save online managing of multisig wallets with geographically distributed keys could be a great contribution for future security and single point of failure resistance!

**Comment by Funman2** [comment]
Just for your information 2 more things I found while testing: 

1.) When installing on Windows, it is kind of a "standalone" app (won't show in the installed app list). I guess this is on purpose, but it does not integrate in windows, so I have to launch the app from shortcut within the extracted folder, when creating my own shortcut for tastbar for instance, it will launch, but not show the Nunchuk symbol in taskbar anymore, the shortcut for opening app is kind of strange because it only works within the extracted folder and can't be moved, so this is a little impractical for daily usage...

2.) Latest Linux Ubuntu version (Ubuntu 22.04.1 LTS) does not seem to support  AppImage format anymore, so a different way of compiling could be needed maybe, I coud not start Nunchuk there, no matter how much I tried by googeling workarounds, so only on Ubuntu 20.04.4 LTS (focal fossa) it worked right away with latest official Nunchuk release.

**Comment by Funman2** [comment]
Congratulations Nunchuk! I got little time today to do first tests with desktop and mobile app for collab signing with airgapped key (seedsigner), so far it worked flawless! Awesome job you did here, that´s now a very sophisticated wallet to use for secure multisig.
(At first i had some problems with the desktop app, because it did not take over my data from the previous nunchuk version and i could not log in, also guest account was kind of missing, but then I erased out all old data from the nunchuk folder and could log in to my account with no problems. Also when I start the messenger with the desktop app first, then mobile,setting up the collab wallet, I get no problems with encryption at all as well and signing really works like a charm now.
Tomorrow I will do some further testing and explore guest mode again.)

I am very exited for all the great options in collab multisig now! This wallet is the future. Thx for all your great contributions to the bitcoin space!

**Comment by Funman2** [comment]
Hey, recently I had some extra time to do a bit further testing with the desktop app on windows.
You really implemented a lot of great new features and improvements! However i also found some things which still may not work as intended.

These are the following list of potential bugs I made, hoping this might be useful for further improvements:

1.) The app is installed and integrated now into my windows 10 in a great way! All icons show etc., but now my old keys and wallets are not recognized by the app in guest mode any more. I would have to set up everything from scratch ( had a lot of different wallets there). This might be related to the Database Encryption Feature, which is not working anymore again. meaning nunchuk does not accept an encryption passphrase anymore, after setting password, just my keys disappear
(I noticed the encryption symbol is by default toggle "on", which should be opposite in settings/database encryption)

2.) The stay signed in option is working fine now with email accounts, however when using primary key, the stay signed in does not work. By the way, i just love the Primary Key feature, that's really most innovative and very forward thinking!

3.) Setting up a Primary key with software key worked really like a charm, but when I tried to use my trezor one as a primary key device, it was recognized, i could add the key, but it did not accept this hardware key as a primary key. ( I think as I understood u could use software and hardware keys but no airgapped, right?) 

4.) Also regarding the trezor in general: when importing the key is possible, it asks for pin, then for passphrase, but when i try to enter an actual valid passphrase which would change the fingerprint, it just turns off passphrase in trezor device, so i can not use a trezor one with entering the passphrase through nunchuk? ( I know this device is not optimal, since passphrase should never be entered on app, rather on device ...)

5.) I was able to delete/remove the Primary key easily, without any warning to do not so from nunchuk, from the account, which is not optimal, since then i could not enter the account and had a very hard time restoring it again...maybe this should not be possible so easily...
Hope this helps make nunchuk even more popular in the future, it´s getting really more awesome with tapsigner and all other improvements!

**Comment by tongvanlinh** [comment]
@Funman2 really thanks for your feedback, let us check it one by one. Some issue we still working on them, some new from you reported, we will focus on them with higher priority

**Comment by Funman2** [comment]
Hey, for now I unfortunately had very little time. As far as I can tell, primary key stay signed in works fine now on windows app. 
But the encryption feature does fail, since it does accept the encryption key once and the keys also got encrypted, I doublechecked in  the nunchuk data folders, but then when logging in it does not ask for encryption passphrase anymore ( the "enter the dragon " screen is missing. Also the stored keys are not displayed in the account anymore, since encryption is not decrypted.
The encryption button in preferences now show the "toggle off" initially in the correct way, but when toggling on and setting an encryption password, after logging off and logging in again it still shows encryption as deactivated ( toggle off) there might be some kind of bug here.
Also when changing the encryption passphrase a second time I get error messages, even after completely erasing the apps folders and uninstalling/reinstalling the app, the error stays and i cant set encryption.
![Encryption Error](https://user-images.githubusercontent.com/105175715/198100021-be7142c9-06d1-41a3-809f-cc6752054754.jpg)

**Comment by tongvanlinh** [comment]
@Funman2 There 2 issues as you mentioned here:
1. but then when logging in it does not ask for encryption passphrase anymore
2. The encryption button in preferences now show the "toggle off" initially in the correct way, but when toggling on and setting an encryption password, after logging off and logging in again it still shows encryption as deactivated

We're debuging on db encrption now, when we have fixed version we will let you know

**Comment by Funman2** [comment]
Hey,
found at least 2 more issues in desktop app: when syncing a 2/3 multisig with airgapped seedsigner keys in messenger, sometimes the tx can not be signed in messenger, only after a restart of the windows app it shows in wallet screen, there it can be signed but within messenger, the transaction is shown with pending signatures,but when trying to sign it gives an error " tx not found"...only in the wallet screen I was able to sign it.

Sometimes the app freezes when loading a new wallet with a lot of transaction history. But When restarting, all Transactions show up, so that's not bothering so much.

 also sometimes when setting up a new collab wallet the finalization only syncs when restarting app.

But, overall everything is its a lot better now, airgapped keys work fine with messenger now!

**Comment by tongvanlinh** [comment]
Thank you very much @Funman2 . Your effort is very important for us improve our application. We got your point and try to fix it soon

**Comment by Funman2** [comment]
Welcome, huge fan of nunchuk here!
When finally everything works properly as intended, it is going to be the superior wallet technology and the most attractive user experience.
Thx for building

**Comment by Funman2** [comment]
Hey , in new windows version 1.9.14 encryption still does not work on my windows, even after erasing all data , uninstalling and reinstalling. It does not accept an encryption key, only once then it does not ask for entering password when logging in again.
since the inheritance update, the windows app is not even found for download on your homepage anymore...the concept of nunchuk is great and visionary but I seriously ask myself:
why should anyone want to pay for an app with so many bugs, which due to the many errors is not usable at this stage? Sorry to say but this is the stage of nunchuk ATM.

**Comment by tongvanlinh** [comment]
@Funman2 sorry about this, we're focusing to add more feature and still working on db with encryption., We found some suspicious points related to issue you report, next week we will make pre release version and share you, because we need to fix this with less impact to current user data.

**Comment by tongvanlinh** [comment]
@Funman2 Could you help me try [this version](https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.15/nunchuk-window-1.9.15.zip) with encryption problem.

**Comment by Funman2** [comment]
Hey thanks!  I will try tonight in detail and on new year's... sorry for the criticism.
First test it also showed encryption as switched off though by default and after restart.
After setting password it also did not ask for decryption password after restarting app, this was on guest account for first try.

 Am away from home so tried in different windows machine this time. But so far same error as I had at home.
This is particularly strange as in prior versions you already had encryption working fine.

This was when you still had the stand alone app, not the windows installed app...
Thanks for your quick responses though!



Von meinem/meiner Galaxy gesendet



-------- Ursprüngliche Nachricht --------
Von: Baron Tong ***@***.***>
Datum: 30.12.22 13:24 (GMT+01:00)
An: nunchuk-io/nunchuk-desktop ***@***.***>
Cc: Funman2 ***@***.***>, Mention ***@***.***>
Betreff: Re: [nunchuk-io/nunchuk-desktop] Database encryption does not work when signed in (Issue #20)


@Funman2<https://github.com/Funman2> Could you help me try this version<https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.15/nunchuk-window-1.9.15.zip> with encryption problem.

**Comment by tongvanlinh** [comment]
@Funman2 Thank you very much for reporting our problem. We are still trying to take over and fix them. There are still some functions that have not been developed yet, so we are also trying to speed up the progress.

I think this one" This was when you still had the stand alone app, not the windows installed app..." can help us something. Maybe the userdata's path was different after we make installer for application instead of running through the portable version like before

I will focus on this issue on next week. 
Happy new year !!!. Take relax for holidays

**Comment by Funman2** [comment]
Thanks, will do!

Yes the nunchuk  team always seems to focus on  developing a lot of thoughtful and leading new features for nunchuk in quick order but in my eyes, maybe you should also sometimes focus on perfecting the basic features and make them more error free for collaborative multisig, which is your core feature!
 If this does not work in a secure way (For me this would be including encryption of keys on desktop) I am still hesitating to use it for securing my major stash... otherwise it would be really perfect, since i would feel a lot safer with geographically distributed keys combined with the ease of use on nunchuk collab wallets.

Happy new year to you as well! Hope I will find time to do some testing and playing with multisig setups on nunchuk. This is indeed relaxing on my free time for me, since my real world job has nothing to do with IT at all;-)!
Take care



Von meinem/meiner Galaxy gesendet



-------- Ursprüngliche Nachricht --------
Von: Baron Tong ***@***.***>
Datum: 30.12.22 15:53 (GMT+01:00)
An: nunchuk-io/nunchuk-desktop ***@***.***>
Cc: Funman2 ***@***.***>, Mention ***@***.***>
Betreff: Re: [nunchuk-io/nunchuk-desktop] Database encryption does not work when signed in (Issue #20)


@Funman2<https://github.com/Funman2> Thank you very much for reporting our problem. We are still trying to take over and fix them. There are still some functions that have not been developed yet, so we are also trying to speed up the progress.

I think this one" This was when you still had the stand alone app, not the windows installed app..." can help us something. Maybe the userdata's path was different after we make installer for application instead of running through the portable version like before

I will focus on this issue on next week.
Happy new year !!!. Take relax for holidays

**Comment by tongvanlinh** [comment]
@Funman2 I'm back and now I stay on this issue. I've update latest portable version, you spend little bit effort help me check this. I'm guessing the db path was difference 
[PORTABLE VERSION](https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.15/nunchuk-window-1.9.15_portable.zip)

**Comment by Funman2** [comment]
Sure of course!

**Comment by Funman2** [comment]
Dear Baron Tong,
So I tried both new versions of nunchuk-window-1.9.15  now a bit and had quite good experience and interesting findings:
So I am back home now on my personal machine and first surprise: 
Even the installed version is working fine now for encryption again on guest account! Also the portable version works for this. When I start up the app and want to log on to guest mode it prompts for " enter the dragon", after that wallets are restored, so on. So this is solved fine!

However, if I then further want to log into an account using my primary key, it still asks for the initial encryption password, this does not seem as intended, right? So when I then enter the set passphrase, it still logs in to guest mode first, then I have to do a second attempt to sign in primary key...

Then I tried to log in to my nunchuk email based account and it worked right away from starting up hte app, without prompting me the encryption password. I could log on swa my testwallet and messenger including all messages. But when i tried to set another encryption password for this account ( wich i am not sure i need) it still fails to reload the keys and wallets after loging out/in again, then the account seems empty until i delete the encrypted data from the nunchuk user data folder in windows...
I hope that helps!
thanks again for putting in so much effort  towards enabling me to selfcustody savely with multisig!

**Comment by tongvanlinh** [comment]
Thx @Funman2 . I will continue to check with account (email base) and primary account for same situation (enable encryption db). your info is helpful with us, really thank, I will feedback soon

**Comment by Funman2** [comment]
Yeah most welcome!
Appreciate the project very much.
By the way is there any btc donation address for the project or a crowdfunding sort of thing?



Von meinem/meiner Galaxy gesendet



-------- Ursprüngliche Nachricht --------
Von: Baron Tong ***@***.***>
Datum: 03.01.23 10:46 (GMT+01:00)
An: nunchuk-io/nunchuk-desktop ***@***.***>
Cc: Funman2 ***@***.***>, Mention ***@***.***>
Betreff: Re: [nunchuk-io/nunchuk-desktop] Database encryption does not work when signed in (Issue #20)


Thx @Funman2<https://github.com/Funman2> . I will continue to check with account (email base) and primary account for same situation (enable encryption db). your info is helpful with us, really thank, I will feedback soon

**Comment by Funman2** [comment]
so, did some further testing and had strange discovery: setting up a primary key ( importing seed+ passphrase) finally worked after erasing all previous data, then setting up primary key first, after that setting encryption for guest mode, then after that, when trying to sign in with primary key, nunchuk prompts me for passphrase first ( although I did not encrypt primary account) , after it gets some error messages: "could not change password ..."or something very briefly but then signs in successfully...

befor it always crashed after setting encryption in primary mode, this time only guest mode has password. Hope that info is helpful as well.

**Comment by tongvanlinh** [comment]
@Funman2 thx you very much. Actually I also saw that problem, now i'm trying with some change. It will take some days and verify, then I'll share soon

**Comment by Funman2** [comment]
Hey, after some further testing I found: 
When signed in with desktop app to a nunchuk account, an airgapped key transaction signature does not sync to mobile apps ( android) , where as the signatures from mobile do sync with desktop.

**Comment by tongvanlinh** [comment]
@Funman2  thank for your effort, I will check it on next week, Now I'm on encryption db issue, we already know problem and we're on testing it. next week I will share new patch

**Comment by tongvanlinh** [comment]
@Funman2 I've new patch for encryption db issue, could you help me check with your desktop ? I already have some round for testing with our db well.
[1.9.15.1](https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.15/nunchuk-window-1.9.15.1.zip)

**Comment by tongvanlinh** [comment]
@Funman2  about this. Could you check this setting on desktop ? (It should be enabled)
![image](https://user-images.githubusercontent.com/12754994/211338817-f804d1ea-7173-4f7c-8e2f-03818aeff124.png)

**Comment by tongvanlinh** [comment]
@Funman2  about this one : ```When signed in with desktop app to a nunchuk account, an airgapped key transaction signature does not sync to mobile apps ( android) , where as the signatures from mobile do sync with desktop.```

Could you check this setting on desktop ? (it should be enable)
![image](https://user-images.githubusercontent.com/12754994/211339401-022219b0-eda6-4bad-a121-5a3f8e167101.png)

**Comment by Funman2** [comment]
Hey of course I will check once I am home. At work I only carry my mobile. Looking forward!  Thx



Von meinem/meiner Galaxy gesendet



-------- Ursprüngliche Nachricht --------
Von: Baron Tong ***@***.***>
Datum: 09.01.23 16:04 (GMT+01:00)
An: nunchuk-io/nunchuk-desktop ***@***.***>
Cc: Funman2 ***@***.***>, Mention ***@***.***>
Betreff: Re: [nunchuk-io/nunchuk-desktop] Database encryption does not work when signed in (Issue #20)


@Funman2<https://github.com/Funman2> about this one : When signed in with desktop app to a nunchuk account, an airgapped key transaction signature does not sync to mobile apps ( android) , where as the signatures from mobile do sync with desktop.

Could you check this setting on desktop ? (it should be enable)
[image]<https://user-images.githubusercontent.com/12754994/211339401-022219b0-eda6-4bad-a121-5a3f8e167101.png>

**Comment by Funman2** [comment]
> 

Hey, so success! Signing in with  Primary Key now works right away when starting up the app. Database Encryption also worked now! Awesome. 
Although after setting up the encryption first attempt  to log in the Nuchuk app crashed after logging in. Then second time it was freezing for 30 Seconds, then after that logging in went smooth the other times!
Also changing the encryption password works now, which sometimes was an issue in prior version. Well done.
For syncing airgapped keys i will test now.

**Comment by Funman2** [comment]
Hey did some further testing, the signed transaction does not import the signature in messenger to mobile app still. Even after enabling multi-device sync option in all devices, but i now ahve a lot of chaos of many wllets and keys all over the place;-)

Signed transaction imports fine from mobile app into desktop version of nunchuk via messenger. 

Also i found after setting encryption in primary account the app keeps crashing avery second time after log in or keeps freezing while syncing, then is usable after maybe one minute...

**Comment by tongvanlinh** [comment]
@Funman2 in case of encryption db, it will take time to decrypt (we're improve this point), 
Can you share video in the crashes issue?

**Comment by Funman2** [comment]
https://user-images.githubusercontent.com/105175715/211508051-8b4a706e-9317-409c-9c67-e20cc65795e5.mp4

Hey, so the crashing got a lot better when I erased DB and keys and set up the collab. wallet from scratch in my primary account ( importing the airgapped keys to all devices then finalize...this way i only can reproduce the crash after logging out of my primary account and trying to log in again, then crash as shown in video....

**Comment by tongvanlinh** [comment]
@Funman2 thank you very much, I will check it up

**Comment by tongvanlinh** [comment]
@Funman2  This is new patch for crashes issue. Please help me check this.
[1.9.15.2](https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.15/nunchuk-window-1.9.15.2.zip)

**Comment by Funman2** [comment]
Thank you for posting this version. I tried but could not find a difference from the previous version.
But signing into the account works fine every time without crash after restarting the app. Only when signing out and then try to sign in again, without a restart of Nunchuk, the crash happens. Not really bothering me though.

**Comment by Funman2** [comment]
Little detail I noticed on new version: when signing in with email account, I now get 2 nice little message "please give some time to sync" and " merging the data". Whereas when signing in with primary account this message is not shown.
Biggest issue still is that collab signatures still do not sync with mobile app with airgapped keys. They only get imported into windows app coming from mobile but not the other direction.
Also if i might say, it still  would be nice if you just could always hit enter ( return) after typing the passphrase and password for signing into the account, still have to click there...

**Comment by tongvanlinh** [comment]
@Funman2 thank for your effort, I having some improment on this one, I will share new patch soon. I really appreciate your contribution

**Comment by tongvanlinh** [comment]
@Funman2 We've new version which is preparing for next release. Can you have some experiment for a bit? and please check it with siging with air gap key also
[DOWNLOAD](https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.16/nunchuk-window-1.9.16.zip)

**Comment by Funman2** [comment]
For sure!  Will do that, gonna test this evening and get back to you.

**Comment by Funman2** [comment]
Hey, I had half an hour to start a little testing already. So far , i got stuck on a new issue: When trying to sign in to a primary account, the order of entering the Passphrase for encryption has changed, now it´s first the encryption passphrase, then the after that I had to enter my passphrase for the primary key. So now it crashes every time after entering this second password, i can´t even log on to the account any more. Will eventually make a short video again.

**Comment by Funman2** [comment]
[screen-capture (16).webm](https://user-images.githubusercontent.com/105175715/212688320-7c3a51fb-d918-4a12-b80b-35405d1b7d28.webm)

**Comment by tongvanlinh** [comment]
@Funman2 thank for your feedback, I will check crashes more.

About the order of inputing passphrase: we change it because db need to be decrypted before login (to get its info)

**Comment by tongvanlinh** [comment]
@Funman2 I've fix it, please take it for crashes one. [DOWNLOAD](https://github.com/nunchuk-io/nunchuk-desktop/releases/download/1.9.16/nunchuk-window-1.9.16.zip)

**Comment by Funman2** [comment]
So got good and bad news: 
The crash is issue is solved now with encrypted account.

But still the airgapped signature does not export to mobile andoid app via messenger. 
Only the android signed transaction imports to desktop, where I can get enough signers by signing here and importing from mobile via messenger. But the signature from desktop still does not export to android.

**Comment by Funman2** [comment]
Wait I rebuild the collab wallet from scratch and rescanned the airgapped keys, then it worked. Now the signatures do sync in both ways from desktop to android and vice versa!
So this way it works now! Good job!

**Comment by Funman2** [comment]
found another small bug: now after deleting wallets and keys in primary account and setting up again, the sync with electrum seems to be stuck, since it always shows  status syncing and never finshes, but in other unchanged account it finishes sync quickly and shows status online.

**Comment by tongvanlinh** [comment]
@Funman2  about airgap key signing. It still occurs on old transaction or new transaction ?

**Comment by Funman2** [comment]
Could not do much testing yesterday, but with new freshly setup account the signing worked!
On my old setup it did not work first with new transaction...
Will do further  testing this morning.

**Comment by Funman2** [comment]
Hey after further testing I can say it totally works now! Awesome.
I did an uninstall of nunchuk desktop app and reinstall, then setting up primary key account from by importing the seed again, setting up the keys and the collab multisig wallet. So with this fresh database it works flawless without any errors now.
I guess the problem was somehow about imported keys of my old database somehow which did not sync. But now also electrum server syncs and transactions can be signed collaborative. So finally this turns into a totally secure and usable wallet not for me, I will consider hodling with this setup in the future.
Also signing out and signing in again works now without crash. Guest mode works fine.
For the Android app: It really crashes all the time when I try to log in to primary, multiple times, then eventually it works. It´s a little better after reinstalling but not optimal. Should I report this as a moblie issue for the android repository?Also adding a second primary account is tricky since you only could do using a trick, since once primary is set up, it only allows you to log in but not add another existing primary account, only when aproaching this process from guest mode it worked somehow sometimes...
Great work overall though!

**Comment by tongvanlinh** [comment]
@Funman2 I really thank you for your positive contribution to nunchuk's product.  We still continue to optimize and develop better functions for the best and safest of user experience.  
Regarding android if possible, you can create a new issue, we will handle for android version soon as well

**Comment by Funman2** [comment]
Android is stable now after latest release. Cool!
So my setup now actually works. What an achievement for security in multisig!
Jjust noticed the desktop app works fast and fluent in collab. mode without encryption but with the database encryption activated, it slows down notably much and after signing in it still takes about a minute or two until the wallet is in a usable state while syncing. In this time it tends to freeze for some period sometimes, but no crashes anymore.

**Comment by Funman2** [comment]
Hey, I just downloaded and tested new official windows version (nunchuk-windows-v1.9.16) +new android version
Everything works absolutely great and perfect now together with android devices and other desktop machines! Seedsigner/ primary account ,2 of 3 multisig, sync on multiple devices works perfect! Great job! 
If I might side-note: Was so totally happy to sign in email accounts now by just hitting enter key ;-)! it would be totally awesome to get the same after entering the encryption passphrase and the passphrase for primary keys...there still you have to mouse click. Thanks for building such a useful tool!

**Comment by Funman2** [comment]
Si since this issue is solved so well this could be closed? ( new issue i posted seperately #31 )
